packages/react/src/time-picker/TimePicker.test.tsx

```diff
@@ -0,0 +1,74 @@
+import '@testing-library/jest-dom';
+import 'html-validate/jest';
+import { axe, toHaveNoViolations } from 'jest-axe';
+import userEvent from '@testing-library/user-event';
+import { render, cleanup, act, waitFor } from '../../../../test-utils';
+import { TimePicker, type TimePickerProps } from './TimePicker';
+
+expect.extend(toHaveNoViolations);
+
+afterEach(cleanup);
+
+function renderTimePicker(props?: Pick<TimePickerProps, 'loading'>) {
+	return render(
+		<TimePicker
+			label="Select a time"
+			hint="Start typing to see results"
+			{...props}
+		/>
+	);
+}
+
+describe('TimePicker', () => {
+	it('renders correctly', () => {
+		const { container } = renderTimePicker();
+		expect(container).toMatchSnapshot();
+	});
+
+	it('renders correctly when loading', async () => {
+		const { container } = renderTimePicker({ loading: true });
+		const input = container.querySelector('input');
+		expect(input).toBeInTheDocument();
+		if (!input) return;
+
+		await act(async () => input.click());
+		expect(container).toMatchSnapshot();
+	});
```

Comment:
We could improve a few things here if we lean on jest/testing-library: returning `undefined` results in a test passing, `getByRole` plus `userEvent` add accessibility validation and more closely resemble user interaction.

Let me know what you think of this:

```suggestion
	it('renders correctly when loading', async () => {
		const { container } = renderTimePicker({ loading: true });
		const input = screen.getByRole('combobox', {
			name: 'Select a time (optional)',
		});
		const user = userEvent.setup();

		await user.click(input);

		expect(container).toMatchSnapshot();
	});
```

Reply:
* you'd need to import `screen` from `test-utils` or modify the suggestion to make it work.

Reply:
Was just copying combobox's tests. If this is cleaner/better, then all for it